add note for URL encoding db connection strings (#3383)
mattermost/docs: source/administration/config-settings.rst

.. note::
   If a setting is set through an environment variable and any other changes are made in the System Console, the value stored of the environment variable will be written back to the ``config.json`` as that setting's value.
   
.. warning::

>    Database connection strings for the database read and search replicas need to be formatted using `URL encoding <https://www.w3schools.com/tags/ref_urlencode.asp>`. As some characters may terminate the string early and will cause issues when the connection string is parsed.

Reviewer: ```suggestion
   Database connection strings for the database read and search replicas need to be formatted using `URL encoding <https://www.w3schools.com/tags/ref_urlencode.asp>`__. Incorrectly formatted strings may cause some characters to terminate the string early, resulting in issues when the connection string is parsed.
```
